core/sql/generator/GenPreCode.cpp

```diff
@@ -11274,6 +11274,485 @@ short HbaseAccess::extractHbaseFilterPreds(Generator * generator,
+// To push down, the predicate must have the following form:
+//  xp:=  <column>  <op>  <value-expr>
+//  xp:=  <column> is not null (no support for hbase lookup)
+//  xp:=  <column> is null (no support for hbase lookup)
+// 	  (xp:=<column> like <value-expr> not yet implemented)
+//  xp:=<xp> OR <xp> (not evaluated in isHbaseFilterPredV2, but by extractHbaseFilterPredV2)
+//  xp:=<xp> AND <xp>(not evaluated in isHbaseFilterPredV2, but by extractHbaseFilterPredV2)
+//
+// and all of the following conditions must be met:
+//
+//      <column>:       a base table or index column which can be serialized and belong to the table being scanned.
+//                            serialized: either the column doesn't need encoding, like
+//                                            an unsigned integer,  or the column
+//                                            was declared with the SERIALIZED option.
+//						it also must not be an added column with default non null.
+//      <op>:              eq, ne, gt, ge, lt, le
+//      <value-expr>:  an expression that only contains const or param values, and
+//                     <value-expr>'s datatype is not a superset of <column>'s datatype.
+//
+/////////////////////////////////////////////////////////////////////////////
+NABoolean HbaseAccess::isHbaseFilterPredV2(Generator * generator, ItemExpr * ie,
+					 ValueId &colVID, ValueId &valueVID,
+					 NAString &op)
+{
+  NABoolean foundBinary = FALSE;
+  NABoolean foundUnary = FALSE;
+  NABoolean hbaseLookupPred = FALSE;
+  NABoolean flipOp = FALSE;  // set to TRUE when column is child(1)
+
+  if (ie &&
+      ((ie->getOperatorType() >= ITM_EQUAL) &&
+       (ie->getOperatorType() <= ITM_GREATER_EQ))) //binary operator case
+    {//begin expression
+      ItemExpr * child0 = ie->child(0)->castToItemExpr();
+      ItemExpr * child1 = ie->child(1)->castToItemExpr();
+
+      if ((ie->child(0)->getOperatorType() == ITM_BASECOLUMN) &&
+	  (NOT hasColReference(ie->child(1))))
+	{
+      foundBinary = TRUE;
+	  colVID = ie->child(0)->getValueId();
+	  valueVID = ie->child(1)->getValueId();
+	}
+      else if ((ie->child(1)->getOperatorType() == ITM_BASECOLUMN) &&
+	       (NOT hasColReference(ie->child(0))))
+	{
+      foundBinary = TRUE;
+      flipOp = TRUE;
+	  colVID = ie->child(1)->getValueId();
+	  valueVID = ie->child(0)->getValueId();
+	}
+      else if ((ie->child(0)->getOperatorType() == ITM_INDEXCOLUMN) &&
+	       (NOT hasColReference(ie->child(1))))
+	{
+      foundBinary = TRUE;
+	  colVID = ie->child(0)->getValueId();
+	  valueVID = ie->child(1)->getValueId();
+	}
+      else if ((ie->child(1)->getOperatorType() == ITM_INDEXCOLUMN) &&
+	       (NOT hasColReference(ie->child(0))))
+	{
+      foundBinary = TRUE;
+      flipOp = TRUE;
+	  colVID = ie->child(1)->getValueId();
+	  valueVID = ie->child(0)->getValueId();
+	}
+      else if ((ie->child(0)->getOperatorType() == ITM_REFERENCE) &&
```

Comment:
You used this reason of it being in V1 repeatedly, and it's understandable, but I hope you won't repeat errors just because there were there before. Most of our code will show some errors and inefficiencies if we do a review of it. Also, ultimately we should get rid of the V1 code.